### ISSUE TITLE
Reorganize source file tree

### ISSUE THREAD

**Issue by jar600** [probcomp/metaprob-clojure#24]
Goal: make the structure immediately clear(er) to new people arriving at the source code.

 - Put all clojure (non-metaprob) source (trace, sequence, builtin_impl) in one subdirectory of `src/metaprob`, say `kernel` or `core`
     - Rename builtin_impl.clj and/or split it up into smaller files
     - Not sure where syntax.clj goes, kernel I guess?
 - Put all metaprob source (prelude, infer, distributions) in another subdirectory, say `metaprob` or `prelude`, or maybe leave it where it is
 - Rename builtin.clj to kernel.clj maybe, leave it in the src/metaprob directory?

**Comment by jar600** [comment]
I'm putting this off pending advice from @zane regarding project module packaging - not exactly the same problem, but I want to know the bigger packaging picture before starting a reorg.

**Comment by jar398** [comment]
Would love to hear ideas from @zane as soon as he is ready with some. Links to external articles or documentation would qualify.

**Comment by zane** [comment]
@jar398 I'd be happy to share some thoughts. Just so I understand: You're wanting to know my opinions regarding what source file tree would be the most intuitive to me as a Clojure programmer?
